src/engraving/dom/textbase.cpp

```diff
@@ -741,19 +741,36 @@ int TextCursor::position(int row, int column) const
 //   TextFragment
 //---------------------------------------------------------
 
-TextFragment::TextFragment()
+TextFragment::TextFragment(const muse::modularity::ContextPtr& iocCtx)
+    : muse::Injectable(iocCtx)
 {
 }
 
-TextFragment::TextFragment(const String& s)
+TextFragment::TextFragment(const String& s, const muse::modularity::ContextPtr& iocCtx)
+    : TextFragment(iocCtx)
 {
     text = s;
 }
 
-TextFragment::TextFragment(TextCursor* cursor, const String& s)
+TextFragment::TextFragment(TextCursor* cursor, const String& s, const muse::modularity::ContextPtr& iocCtx)
+    : TextFragment(s, iocCtx)
 {
     format = *cursor->format();
-    text = s;
+}
+
+TextFragment::TextFragment(const TextFragment& f)
+    : muse::Injectable(f.iocContext())
+{
+    text = f.text;
+    format = f.format;
+}
+
+TextFragment& TextFragment::operator =(const TextFragment& f)
+{
+    text = f.text;
+    format = f.format;
+    pos = f.pos;
+    return *this;
```

Comment:
Why are these necessary?

Reply:
Inject does not have a copy operator, so a compiler cannot create a default implementation of the copy operator